ps-modules/MacUtil/MacUtil.psm1

```diff
@@ -0,0 +1,202 @@
+Import-Module "$PSScriptRoot/../../ps-modules/Util"
+
+function ConvertTo-UniversalBinaries {
+    param (
+        [string]$arm64Dir,
+        [string]$x64Dir,
+        [string]$universalDir
+    )
+
+    Write-Host "Creating universal dir and copying headers..."
+    if (Test-Path -Path $universalDir -PathType Container) {
+       Remove-Item -Path "$universalDir" -Recurse -Force | Out-Null
+    }
+    New-Item -Path "$universalDir" -ItemType Directory -Force | Out-Null
+    $arm64LibDir = Join-Path $arm64Dir "lib"
+    $x64LibDir = Join-Path $x64Dir "lib"
+    Copy-Item -Path "$x64Dir\include" -Destination "$universalDir\include" -Recurse | Out-Null # Assume arm64 and x86_64 are identical
+
+    Write-Host "Making install paths relative..."
+    ConvertTo-RelativeInstallPaths -directory $arm64LibDir -extension "a"
+    ConvertTo-RelativeInstallPaths -directory $arm64LibDir -extension "dylib"
+    ConvertTo-RelativeInstallPaths -directory $x64LibDir -extension "a"
+    ConvertTo-RelativeInstallPaths -directory $x64LibDir -extension "dylib"
+
+    Write-Host "Making binaries universal..."
+    $universalLibDir = Join-Path $universalDir "lib"
+    New-Item -Path "$universalLibDir" -ItemType Directory -Force | Out-Null
+
+    Write-Host "Looking in: $arm64LibDir"
+    $items = Get-ChildItem -Path "$arm64LibDir/*" -Include "*.dylib","*.a"
+    foreach($item in $items) {
+        $fileName = $item.Name
+        $srcPathArm64 = $item.FullName
+        $srcPathX64 = (Join-Path $x64LibDir $fileName)
+        $destPath = (Join-Path $universalLibDir $fileName)
+        
+        if ($item.Attributes -band [System.IO.FileAttributes]::ReparsePoint) {
+            Write-Host "> Processing: $fileName - Copying symlink"
+            #Copy-Item -LiteralPath $srcPathArm64 -Destination $destPath -Force -Recurse
+            Invoke-Expression -Command "cp -R `"$srcPathArm64`" `"$destPath`""
+        }
+        elseif (-not $item.PSIsContainer) {
+            Write-Host "> Processing: $filename - Running lipo"
+            Invoke-Expression -Command "lipo -create -output `"$destPath`" `"$srcPathArm64`" `"$srcPathX64`""
+        }
+    }
+}
+
+function Update-LibraryPath {
+    param (
+        [string]$libPath
+    )
+
+    Write-Host ">> Update-LibraryPath..."
+
+    if (-not (Test-Path $libPath -PathType Leaf)) {
+        Write-Host ">> File '$libPath' does not exist."
+        return
+    }
+
+    Write-Host ">> Updating library path for: $libPath..."
+    #$otoolCmd = "otool -L $libPath"
+    #$otoolOutput = Invoke-Expression -Command $otoolCmd
```

Comment:
Does this function work? Looks like we're not even calling `otool` here?

Reply:
That just displays the dependencies, which was previously used before and after updating the dependencies as debugging info.  I left it in there and commented out, but maybe I should just remove it to avoid confusion.

Reply:
Actually, you were right.  It does need to run beforehand.  Fixed: 5955eb275261c3b2721070988924f61016f8a2e2